packages/@aws-cdk/aws-events/lib/schedule.ts

```diff
@@ -16,7 +16,11 @@ export abstract class Schedule {
   /**
    * Construct a schedule from an interval and a time unit
    */
-  public static rate(duration: Duration): Schedule {
+  public static rate(duration: Duration): Schedule {    
+    if(duration.isUnresolved()){
+      return new LiteralSchedule(`rate(${duration.toHumanString()})`); 
```

Comment:
I believe this [only supports minutes, hours and days](https://docs.aws.amazon.com/eventbridge/latest/userguide/scheduled-events.html#rate-expressions) but `Duration` also supports seconds and milliseconds. It could support more in the future.

Also, I believe the number must not be a fraction

Could we validations for these, and throw an appropriate error.

Reply:
Sure. I see that Seconds should work with Schedule rate since duration.toSeconds() has convert() which would match the unit and not convert if units match. But, milliseconds is left. I can add a check for it being unsupported.Does the Error "Unsupported duration format - millisecond" sound ok? 

Regarding fraction, I saw convert() has a check
```
 if (!Number.isInteger(value) && integral) {
    throw new Error(`'${amount} ${fromUnit}' cannot be converted into a whole number of ${toUnit}.`);
  } 
``` 
Does it help?

Reply:
Regarding seconds, I see in the doc it only mentions minute | minutes | hour | hours | day | days. Let me add check for seconds too.